Client/Frontend/Widgets/SnackBar.swift

```diff
@@ -141,33 +148,51 @@ class SnackBar: UIView {
             return
         }
 
+        let imageHeight: CGFloat
+        let imageWidth: CGFloat
         if let img = imageView.image {
+            imageHeight = img.size.height + AppConstants.DefaultPadding * 2
+            imageWidth = img.size.width + AppConstants.DefaultPadding * 2
             imageView.snp_remakeConstraints({ make in
-                make.left.equalTo(self).offset(AppConstants.DefaultPadding)
-                make.top.equalTo(self).offset(AppConstants.DefaultPadding)
-                make.width.equalTo(img.size.width)
-                make.height.equalTo(img.size.height)
+                make.left.equalTo(self)
+                make.top.equalTo(self)
+                make.size.equalTo(CGSize(width: imageWidth, height: imageHeight))
             })
         } else {
+            imageHeight = 0
+            imageWidth = AppConstants.DefaultPadding
+
             imageView.snp_remakeConstraints({ make in
-                make.width.equalTo(0)
+                // To avoid doubling the padding, the textview doesn't have an inset on its left side.
+                // Instead, it relies on the imageView to tell it where its left side should be.
+                make.width.equalTo(imageWidth)
                 make.height.equalTo(0)
-                make.top.left.equalTo(self).offset(AppConstants.DefaultPadding)
+                make.top.equalTo(self)
+                make.left.equalTo(self)
             })
         }
 
-        let labelSize = self.textView.sizeThatFits(CGSizeMake(self.frame.width, CGFloat(MAXFLOAT)))
-        textView.textContainerInset = UIEdgeInsetsZero
+        // Find what size label will fit given this imageWidth
+        // On wide screen devices, our width will be fixed. That may not be known yet.
+        let snackBarWidth: CGFloat
+        if traitCollection.horizontalSizeClass != .Regular {
```

Comment:
Instead of referencing the superview's width inside the widget, we should let the external user of this widget define constraints on it and let that infer the size instead of setting the width to be fixed internally. In the case of needing a width for the label, I believe there's a way you can have the label automatically resize based on it's contents if you set preferredMaxLayoutWidth = self.frame.size.width. I know there was a bug in iOS 6/7 that required you to do the following:

```
override func layoutSubviews() {
    super.layoutSubviews()
    self.textView.preferredMaxLayoutWidth = self.width.frame.size.width - (imageSize + padding)
   super.layoutSubviews()
} 
```

But I think in iOS 8 this might be resolved so you should be able to just set the preferredMaxLayoutWidth.